sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -512,6 +512,18 @@ class SubquerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       Row(1, null) :: Row(2, 6.0) :: Row(3, 2.0) :: Row(null, null) :: Row(6, null) :: Nil)
   }
 
+  test("SPARK-34252: correlated scalar subquery with view in aggregate") {
+    withTable("tr") {
+      withView("vr") {
+        r.write.saveAsTable("tr")
+        sql("create view vr as select * from tr")
+        checkAnswer(
+          sql("select a, (select sum(d) from vr where a = c) sum_d from l l1 group by 1, 2"),
```

Comment:
Note that the existing test with the same query worked fine because `r` is a dataframe temp view, which doesn't have the `View` node.